src/mlpack/methods/preprocess/preprocess_scale_main.cpp

```diff
@@ -0,0 +1,174 @@
+    PRINT_DATASET("X") + " into " + PRINT_DATASET("X_whitened")+ " with "
+    " PCA as whitening_method and use 0.01 as regularization parameter, "
+    "we coud run "
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X", "output", "X_scaled",
+    "scaler_method", "whitening", "whitening_method", "PCA", "epsilon", 0.01) +
+    "\n\n"
+    "Another simple example where we want to scale the dataset " +
+    PRINT_DATASET("X") + " into " + PRINT_DATASET("X_scaled") + " with "
+    " min_max_scaler as scaler method, where scaling range is 1 to 3 instead"
+    " of deafult 0 to 1. We coud run "
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X", "output", "X_scaled",
+    "scaler_method", "min_max_scaler", "min_value", 1, "max_value", 3),
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+// Define parameters for data.
+PARAM_MATRIX_IN_REQ("input", "Matrix containing data.", "i");
+PARAM_MATRIX_OUT("output", "Matrix to save scaled data to.", "o");
+PARAM_STRING_IN("scaler_method", "method to use for scaling, the "
+    "default is standard_scaler.", "a", "standard_scaler");
+PARAM_STRING_IN("whitening_method", "method to use for whitening, the "
+    "default is ZCA.", "w", "ZCA");
+PARAM_DOUBLE_IN("epsilon", "regularization Parameter, should be between"
+  "-1 to 1.", "r", 0.000001);
+
+PARAM_INT_IN("seed", "Random seed (0 for std::time(NULL)).", "s", 0);
+PARAM_INT_IN("min_value", "Starting value of range for min_max_scaler.",
+    "b", 0);
+PARAM_INT_IN("max_value", "Ending value of range for min_max_scaler.",
+    "e", 1);
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+static void mlpackMain()
+{
+  // Parse command line options.
+  const std::string scalerMethod = CLI::GetParam<string>("scaler_method");
+
+  if (CLI::GetParam<int>("seed") == 0)
+    mlpack::math::RandomSeed(std::time(NULL));
+  else
+    mlpack::math::RandomSeed((size_t) CLI::GetParam<int>("seed"));
+  // Make sure the user specified output filenames.
+  RequireAtLeastOnePassed({ "output" }, false, "no output will be saved");
+  // Check scaler method.
+  RequireParamValue<std::string>("scaler_method",
+      [](std::string x) { return x == "standard_scaler" || x ==
+      "min_max_scaler" || x == "mean_normalization" || x == "max_abs_scaler" ||
+      x == "whitening";},true, "scaler_method must be one among min_max_scaler"
+      ",max_abs_scaler, whitening, standard_scaler or mean_normalization.");
+  // If scaler_method is not set, warn the user.
+  if (!CLI::HasParam("scaler_method"))
+  {
+    Log::Warn << "You did not specify " << PRINT_PARAM_STRING("scaler_method")
+        << ", so it will be automatically set to standard_scaler." << endl;
+  }
+
+  // Load the data.
+  arma::mat input = CLI::GetParam<arma::mat>("input");
```

Comment:
I'd suggest using `std::move()` here, or having `input` be a reference instead.

Reply:
Done.